src/Tools/CsvImporter/Services/MainService/MainService.cs

```diff
@@ -0,0 +1,291 @@
+            FileInfo csvFileInfo = new(relativePathToCsv);
+
+            if (!csvFileInfo.Exists)
+            {
+                FileNotFoundException csvNotFoundException = new("File not found", csvFileInfo.FullName);
+                _logger.LogError(csvNotFoundException, "Error reading CSV file");
+                exitCode = 50;
+
+                throw csvNotFoundException;
+            }
+
+            if (csvFileInfo.Extension != ".csv")
+            {
+                ArgumentException csvFileExtensionException = new("File must be a CSV file", csvFileInfo.FullName);
+                _logger.LogError(csvFileExtensionException, "Error reading CSV file");
+                exitCode = 51;
+
+                throw csvFileExtensionException;
+            }
+
+            _logger.LogInformation("Reading CSV file: {CsvFilePath}", csvFileInfo.FullName);
+
+            // Read CSV file.
+            List<UserDetails> userDetailsList;
+            try
+            {
+                userDetailsList = await CsvFileReader.ReadCsvFileAsync(
+                    logger: _logger,
+                    csvFilePath: csvFileInfo.FullName
+                );
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error reading CSV file");
+                exitCode = 30;
+
+                return;
+            }
+
+            _logger.LogInformation("Found {UserDetailsCount} users in CSV file", userDetailsList.Count);
+
+            // If last run CSV file path exists, 
+            // read the last run CSV file.
+            List<UserDetails>? lastRunUserDetailsList = null;
+            if (_configService.Config.LastCsvPath is not null)
+            {
+                Path.GetRelativePath(
+                    relativeTo: runningDir,
+                    path: _configService.Config.LastCsvPath
+                );
+
+                FileInfo lastCsvFileInfo = new(_configService.Config.LastCsvPath);
+
+                if (lastCsvFileInfo.Exists)
+                {
+                    _logger.LogInformation("Reading last run CSV file: {LastCsvFilePath}", lastCsvFileInfo.FullName);
+                    lastRunUserDetailsList = await CsvFileReader.ReadCsvFileAsync(
+                        logger: _logger,
+                        csvFilePath: lastCsvFileInfo.FullName
+                    );
+                }
+            }
+
+            // If last run CSV file was read,
+            // get the delta between the last run CSV file and the current CSV file.
+            if (lastRunUserDetailsList is not null && lastRunUserDetailsList.Count != 0)
+            {
+                Stopwatch deltaStopwatch = Stopwatch.StartNew();
+
+                _logger.LogInformation("Found {LastRunUserDetailsCount} users in last run CSV file", lastRunUserDetailsList.Count);
+
+                List<UserDetails> deltaList = await CsvFileReader.GetDeltaAsync(
```

Comment:
Should probably pass a cancellation token here. If a premature app shutdown is called, it will continue running until it completes.